internal/credproviders/custom_default_credentials_provider.go

```diff
@@ -0,0 +1,57 @@
+package credproviders
+
+import (
+    "context"
+    "fmt"
+    "os"
+
+    "github.com/aws/aws-sdk-go-v2/aws"
+    "github.com/aws/aws-sdk-go-v2/credentials"
+    "github.com/aws/aws-sdk-go-v2/credentials/ec2rolecreds"
+    "github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
+)
+
+// internal environment variable keys
+const (
+    envAWSAccessKeyID     = "AWS_ACCESS_KEY_ID"
+    envAWSSecretAccessKey = "AWS_SECRET_ACCESS_KEY"
+    envAWSSessionToken    = "AWS_SESSION_TOKEN"
+)
+
+// chainedProvider tries each provider in order until one succeeds
+type chainedProvider struct {
+    providers []aws.CredentialsProvider
+}
+
+func (c *chainedProvider) Retrieve(ctx context.Context) (aws.Credentials, error) {
+    for _, p := range c.providers {
+        creds, err := p.Retrieve(ctx)
+        if err == nil {
+            return creds, nil
+        }
+    }
+    return aws.Credentials{}, fmt.Errorf("no valid credentials found in chained provider")
+}
+
+// CustomDefaultCredentialsProvider returns a provider that tries static env vars first, then IMDS
+func CustomDefaultCredentialsProvider(cfg aws.Config) aws.CredentialsProvider {
```

Comment:
is this the right implementation? FMU the env var based AWS creds are not rotatable. instead of implementing our own cred provider, can we reuse the existing ones?